src/client/Microsoft.Identity.Client/ManagedIdentity/V2/ImdsV2ManagedIdentitySource.cs

```diff
@@ -199,10 +203,60 @@ private async Task<CertificateRequestResponse> ExecuteCertificateRequestAsync(st
                 { OAuth2Header.XMsCorrelationId, _requestContext.CorrelationId.ToString() }
             };
 
+            string attestationJwt = string.Empty;
+
+            // Normalize endpoint
+            string normalizedEndpoint = NormalizeAttestationEndpoint(attestationEndpoint, _requestContext.Logger);
```

Comment:
Pls return an URI directly, since you create one in NormalizeAttestationEndpoint, transform to string and then back to URI.

Reply:
done